### PR TITLE
dev_am into Twilight. dotenv package added and more startup bugs fixed. command_usage. bloop

### PR DESCRIPTION
**Pull request by Andrew-McCall**
Because this adds a dependancy. I made a PR.

- dotenv added.
- folder /data is generated and hecks generation is fixed.
- command_usage added (/usage), uses an unsafe global variable.
- /bloop is now working. Same as main.

General Polish for Prod (twilight branch)

Just been doing a little bit daily, DM for any more questions.

### CODE REVIEW

File: .gitignore
Comment (Andrew-McCall) [inline .gitignore:9]: Empty folders arent pushed. So having a carveout for data didnt do anything. Could add an empty file in /data or do some git wizardry, but now the code creates /data if it doesn't exist.

File: src/luro.rs
Comment (nurahwolf) [inline src/luro.rs:33]: I should change the var to the english spelling of AUTHORISATION before this branch gets popular, just so there is not breaking change in the future. Feel free to add it on yourself, or I'll do it after

File: .gitignore
Comment (nurahwolf) [reply .gitignore:9]: I'm aware empty folders are not pushed, in mainline Luro there is the sample folder however. This pretty much matched my setup there, and once I add `/data/sample` things will be correct. That raises another question though... Maybe it would be better to force include `!/data/sample` and exclude `/data`?

Thoughts?

File: src/config/hecks.rs
Comment (nurahwolf) [inline src/config/hecks.rs:22]: This should all be `[]` for complete sake. Yeah, this code works, but I like consistency :)
```suggestion
            contents = "hecks = []\nsfw_hecks = []\nnsfw_hecks = []\nsfw_heck_ids = []\nnsfw_heck_ids = []".to_string();
```

File: src/interactions/boop.rs
Comment (nurahwolf) [inline src/interactions/boop.rs:78]: Commented out code? Would say drop it if its not needed, though if its something WIP, leaving it in is fine. Your call

File: src/event_handler.rs
Comment (nurahwolf) [inline src/event_handler.rs:19]: Rust pretty much recommends you do NOT do something like this, additionally it is not thread safe. You would want to wrap this in an [ARC](https://doc.rust-lang.org/std/sync/struct.Arc.html) much like Mainline Luro does for it to be thread safe. HOWEVER, this does not fix the fact that static variables should not be modified like this, so the real fix is below.

Extend the Luro struct (The one that contains `application_id`, `http` and such in `main.rs`) and add this variable. Wrap it in [RwLock](https://doc.rust-lang.org/std/sync/struct.RwLock.html) much like some of the other variables, then you can modify it from within Luro's context. The other commands should help you get a grasp of how to achieve that, such as the hecks command.

**Review by nurahwolf — CHANGES_REQUESTED** [review]
Some changes are needed, but I'm happy with the changes to env. I would like you to rework that use of a static variable though, no need to introduce unsafe code for that.

On a positive note, the way you are creating `InteractionResponse` is very, very nice. I'll start updating the commands I made to be in the same format. Works really well!

File: src/luro.rs
Comment (nurahwolf) [inline src/luro.rs:32]: Update this to use the british english spelling of 'Authorisation'. Mistake on my part!

```suggestion
        // Lavalink authorisation, defined by the "LAVALINK_AUTHORISATION" environmental
```

File: src/luro.rs
Comment (nurahwolf) [inline src/luro.rs:33]: Update this to use the british english spelling of 'Authorisation'. Mistake on my part!
```suggestion
        let lavalink_auth = env::var("LAVALINK_AUTHORISATION").expect("No LAVALINK_AUTHORISATION defined: {err}");
```

**Review by nurahwolf — CHANGES_REQUESTED** [review]
Sorry, screwed up one of my previous review comments

File: src/interactions/boop.rs
Comment (nurahwolf) [reply src/interactions/boop.rs:78]: Another one I slightly misselected. First time using Github's review tools, but it's obvious here I meant the entire block of commented out code

File: .gitignore
Comment (Andrew-McCall) [reply .gitignore:9]: Either would work, But I think just !/data/sample once/if we make it.
Then no custom secrets would be made visible by accident.

File: src/config/hecks.rs
Comment (Andrew-McCall) [reply src/config/hecks.rs:22]: I agree completely, I just did [] since it was what the code already expected. I was only fixing bugs rather than polish. I'm happy to fix it, now I have more experience with the codebase.

File: src/interactions/boop.rs
Comment (Andrew-McCall) [reply src/interactions/boop.rs:78]: I'm not use to github UI either. It was a reminant of my testing. Allows updates without response/other triggers. But I'll remove it in my polish commit before merge. :)

File: src/event_handler.rs
Comment (Andrew-McCall) [reply src/event_handler.rs:19]: I actually edited the luro struct at first but couldn't get it to work. Since it was the first command I made I was hoping to spend more time debugging my twilight implentation instead of my own code. I like your standards and the link is helpful. I'll bump the priority of fixing this.

**Comment by Andrew-McCall** [comment]
Implemented all the changes. RwLock took longer than I'd like to admit since the interaction write was in scope of the read so it hung indefinitely...lol 

The rest was simple enough. 

Comment or PM me for anything.

File: .gitignore
Comment (nurahwolf) [reply .gitignore:9]: Sounds good to me!

File: src/main.rs
Comment (nurahwolf) [inline src/main.rs:74]: Luro's context is already wrapped in an Arc, you don't need it here. My previous comment was more of a note about thread safety. That said, you should just need to accept this suggestion and be fine :)

```suggestion
    interaction_count: tokio::sync::RwLock<u32>,   
```

File: src/luro.rs
Comment (nurahwolf) [inline src/luro.rs:121]: Likewise with my previous suggestion, no need to wrap this in an Arc.
```suggestion
                interaction_count: tokio::sync::RwLock::new(0),
```

**Review by nurahwolf — CHANGES_REQUESTED** [review]
Few minor tweaks, then good to merge!

File: src/luro.rs
Comment (Andrew-McCall) [reply src/luro.rs]: Done

**Pull request merged**
